core/main/src/firebolt/handlers/capabilities_rpc.rs

```diff
@@ -140,19 +140,17 @@ impl CapabilityServer for CapabilityImpl {
             .check_cap_role(&ctx.app_id, cap.clone())
         {
             return Ok(v);
-        } else {
-            if let Ok(_) =
-                PermissionHandler::fetch_and_store(self.state.clone(), ctx.clone().app_id).await
+        } else if let Ok(_) =
+            PermissionHandler::fetch_and_store(self.state.clone(), ctx.clone().app_id).await
```

Comment:
Here as well. We can get a string from app_id rather than cloning context and getting app_id from it.

Reply:
A better approach would be changing the function fetch_and_store to get string slice rather than an owned string.